### ISSUE TITLE
"superclass mismatch for class Schema" if running with ActiveRecord 5.0.0.1

### ISSUE THREAD

**Issue by rammpeter** [nulldb/nulldb#69]
nulldb-adapter is configured as primary database for my rails app in database.yml.
If using nulldb-adapter with rails 5.0.0.1 / active_record 5.0.0.1 it will raise this error with the following stack trace:

TypeError (superclass mismatch for class Schema)

/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activerecord-nulldb-adapter-0.3.5/lib/nulldb/extensions.rb:36:in `<module:ActiveRecord>'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activerecord-nulldb-adapter-0.3.5/lib/nulldb/extensions.rb:34:in`<top>'
org/jruby/RubyKernel.java:944:in `require'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:293:in`block in require'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:257:in `block in load_dependency'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:662:in`new_constants_in'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:257:in `load_dependency'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:293:in`require'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activerecord-nulldb-adapter-0.3.5/lib/active_record/connection_adapters/nulldb_adapter.rb:1:in `<top>'
org/jruby/RubyKernel.java:944:in`require'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:293:in `block in require'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:257:in`block in load_dependency'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:662:in `new_constants_in'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:257:in`load_dependency'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:293:in `require'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activerecord-nulldb-adapter-0.3.5/lib/active_record/connection_adapters/nulldb_adapter.rb:10:in`(root)'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activerecord-5.0.0.1/lib/active_record/connection_adapters/connection_specification.rb:1:in `(root)'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activerecord-5.0.0.1/lib/active_record/connection_adapters/connection_specification.rb:174:in`block in spec'
org/jruby/RubyBasicObject.java:1647:in `instance_eval'
/Users/ramm/.rvm/gems/jruby-9.1.2.0/gems/activerecord-5.0.0.1/lib/active_record/connection_handling.rb:53:in`establish_connection'

**Comment by edwardmp** [comment]
Same issue here on Rails 5

**Comment by edwardmp** [comment]
Think I found the issue;
In Rails 4 and earlier the Schema class inherited from the Migration class.
In Rails 5 this has changed to Migration::Current. Locally modifying the Gem confirmed this fix.

I ask the authors if they would want to see this as a pull request? If so create a seperate branch for rails 5 or include some conditional logic to apply the correct base class depending on the Rails version used.

**Comment by merqlove** [comment]
Hot fix (worked for me before update to 0.3.5):

`gem 'activerecord-nulldb-adapter', github: 'mnoack/nulldb', ref: 'aa36e3c'`

**Comment by mnoack** [comment]
@rammpeter @edwardmp @merqlove The rails5 branch has been merged, and release as 0.3.6, feel free to use the main gem rather than forks again and hopefully as an additional review we can continue to maintain compatibility with future rails versions.

**Comment by edwardmp** [comment]
@mnoack this problem has resurfaced for me in combination with Rails 5.1. Submitted PR to fix.

**Comment by synthead** [comment]
I am also seeing this issue with activerecord 5.1.1 and activerecord-nulldb-adapter 0.3.6.

**Comment by mnoack** [comment]
@synthead can you point your Gemfile to the master branch and see if there's any issues.

I'm just awaiting a bit more feedback from users before I release the next version which supports rails 5.1, rails-edge (5.2).

**Comment by afuno** [comment]
@mnoack Please correct this problem as soon as possible. The project on Rails 5.1 refuses to run:

```
/Users/Colibri/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/bundler-1.14.6/lib/bundler/runtime.rb:94:in `rescue in block (2 levels) in require': There was an error while trying to load the gem 'activerecord-nulldb-adapter'. (Bundler::GemRequireError)
Gem Load Error is: superclass mismatch for class Schema
```

**Comment by mnoack** [comment]
@afuno I was holding off so someone with Rails 5.1 could just confirm that changes that had been done work.

Since I haven't got this and it's effecting enough people I've just made a release anyway and we'll find out.

Thanks for reporting the issue and let me know if there are any issues in the latest release (0.3.7)

**Comment by mnoack** [comment]
@synthead @edwardmp @merqlove @rammpeter - 0.3.7 has been released with rails 5.1/5.2 support. Let me know if there are any issues.

**Comment by edwardmp** [comment]
@mnoack sorry forgot to give feedback earlier. Seems it works great, good job.
